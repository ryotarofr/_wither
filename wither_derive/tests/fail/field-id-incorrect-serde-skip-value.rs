use serde::{Serialize, Deserialize};
use wither::Model;

#[derive(Serialize, Deserialize, Model)]
struct BadModel {
    #[serde(rename="_id", skip_serializing_if="Option::is_some")]
    id: Option<wither::bson::oid::ObjectId>,
}

fn main() {}
