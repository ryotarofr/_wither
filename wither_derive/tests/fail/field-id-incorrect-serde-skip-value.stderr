error: the serde `skip_serializing_if` attr for wither::Model ID fields should be `skip_serializing_if="Option::is_none"`
 --> $DIR/field-id-incorrect-serde-skip-value.rs:6:27
  |
6 |     #[serde(rename="_id", skip_serializing_if="Option::is_some")]
  |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
